### ISSUE TITLE
Notification to hidden (?) Q&A throws error message rather than 404 or similar.

### ISSUE THREAD

**Issue by dajare** [topanswers/topanswers#31]
I'm not sure how "reproducible" this bug will be. I'm getting a notice of 17 messages in a Q&A in which I participated (as flagger/commenter):

<img width="290" alt="image" src="https://user-images.githubusercontent.com/40625/74708721-324e3b00-5215-11ea-923d-bfa323321ef9.png">

The link is to `https://topanswers.xyz/meta?q=718`, the MarkdownIt "toc" bug Q&A which I flagged as "bug > fixed > hide" (comment). 

When I click on the wee icon+notification, I get sent to a white screen with the message:
`ERROR: invalid question CONTEXT: PL/pgSQL function _error(integer,text) line 1 at RAISE SQL function "login_question" statement 3 SQL function "login_question" statement 1`

I assume I'm getting this because my flag has been "accepted", the question "hidden", but I'm still getting notifications and offered a link. Probably, once hidden, the right-sidebar icon should be hidden, too (or, failing that, a more "elegant" `You can't see this!` message).

**Comment by petervandivier** [comment]
Possibly related chat context: https://topanswers.xyz/transcript?room=2&id=177903&year=2025#c177903

TL;DR: static link to a post not visible to the user due to a flag results in error:

> ERROR: invalid question CONTEXT: PL/pgSQL function raise_error(integer,text) line 1 at RAISE SQL function “login_question” statement 2 SQL function “login_question” statement 1

- OP link: https://topanswers.xyz/databases?q=30
- Wayback snapshot: https://web.archive.org/web/20200512050105/https://topanswers.xyz/databases?q=30#a626
